policies/versioning-policy.md

```diff
@@ -0,0 +1,153 @@
+# OpenSSL release versioning policy
+
+This document describes the release versioning scheme used by the OpenSSL
+project from version 3.0.0 onwards.  It also details the level of [ABI] and
+[API] compatibility each version represents.
+
+The version scheme consists a triple of numbers:
+**[major]**.**[minor]**.**[patch]**.
+
+> For example, the version _3.0.1_ has a _major_ version of _3_, a _minor_
+> version of _0_ and a _patch_ version of _1_.
+
+This closely aligns with the expectations of users who are familiar
+with _semantic versioning_.  However, it is not quite _semantic versioning_
+because adopting _semantic versioning_ would mean changing our current
+Long Term Support ([LTS]) policies and practices.
+
+## Major release: changing the first digit
+
+A major release can, and generally will, introduce API and ABI breaking
+changes.
+
+> For example, a program that runs with OpenSSL 3.0.1 should not expect to run
+> with OpenSSL 4.0.0 without modifications.  If it does run, it might not be
+> as efficient as it was.
+
+In order to maintain stability and limit rework across major versions:
+
+- No existing public interface can be modified except where changes
+  are unlikely to break source compatibility or where structures are being
+  made opaque.
+- No existing public interface can be removed until its replacement
+  has been in place in an LTS stable release. The original interface
+  must also have been documented as _deprecated_ for at least 5 years. A
```

Comment:
I think this could do with some clarification. Do we mean "no existing public interface can be removed from *any* release   until the original interface has been documented as deprecated for at least 5 years". Or do we mean "existing public interfaces must remain in at least one stable release until the original interface has been documented as deprecated for at least 5 years".

I prefer the latter interpretation.

Reply:
Not sure I understand the distinction, 'cause surely, we're never going to *remove* stuff from any existing release anyway, right?  Any removal will only happen in a *new major release* anyway, won't it?

What am I missing?

Reply:
I've clarified to the second interpretation.
I've also reworded the prior point to be a positive not a negative statement (for consistency).
I've moved the definition of _public interfaces_ to the definitions document.
I've added a not that examples in the policy are illustrative not definitive.